src/main/java/edu/kis/vh/nursery/FifoRhymer.java

```diff
@@ -0,0 +1,23 @@
+package edu.kis.vh.nursery;
+
+public class FifoRhymer extends DefaultCountingOutRhymer {
+
+    private final DefaultCountingOutRhymer temp = new DefaultCountingOutRhymer();
+
+    @Override
+    public int countOut() {
+        while (!callCheck())
+            getTemp().countIn(super.countOut());
```

Comment:
This is an exaggeration. Using private fields internally usually does not require encapsulation.